File: src/main/java/com/study/platform/global/outbox/model/OutboxEventRepository.java
package com.study.platform.global.outbox.model;

import java.time.LocalDateTime;
import java.util.List;

public interface OutboxEventRepository {

    OutboxEvent save(OutboxEvent event);
    List<OutboxEvent> findAllByStatusAndCreatedAtBefore(OutboxEventStatus status, LocalDateTime createdAt);
    int markSentById(Long id);
    void markFailedPermanently(Long id);
}

File: domain/src/main/kotlin/com/study/platform/global/outbox/model/OutboxEventRepository.kt
package com.study.platform.global.outbox.model

import org.springframework.data.domain.Pageable
import java.time.LocalDateTime

interface OutboxEventRepository {

    fun save(event: OutboxEvent): OutboxEvent
    fun findById(id: Long): OutboxEvent?
    fun findAllByStatusAndCreatedAtBefore(status: OutboxEventStatus, createdAt: LocalDateTime, pageable: Pageable): List<OutboxEvent>
    fun markSentById(id: Long): Int
    fun markFailedPermanently(id: Long)
}
